File: sudoku-common/common-mybatis/src/main/java/com/zumin/sudoku/common/mybatis/excpetion/PageException.java
package com.zumin.sudoku.common.mybatis.excpetion;

import com.zumin.sudoku.common.core.enums.CommonStatusCode;
import com.zumin.sudoku.common.core.exception.BaseException;
import com.zumin.sudoku.common.mybatis.enums.MybatisStatusCode;
import lombok.Getter;

/**
 * 分页异常类
 */
@Getter
public class PageException extends BaseException {

  private static final long serialVersionUID = -1202049169478818925L;

  /**
   * 分页异常类的无参构造方法
   */
  public PageException() {
    super(MybatisStatusCode.PAGE_ERROR);
  }

  /**
   * 分页异常类的构造方法
   *
   * @param statusCode 状态编码
   */
  public PageException(MybatisStatusCode statusCode) {
    super(statusCode);
  }

}

File: sudoku-common/common-mybatis/src/main/java/com/zumin/sudoku/common/mybatis/page/PageException.kt
package com.zumin.sudoku.common.mybatis.page

import com.zumin.sudoku.common.core.exception.BaseException
import com.zumin.sudoku.common.mybatis.MybatisStatusCode

/**
 * 分页异常类
 */
class PageException(
  statusCode: MybatisStatusCode = MybatisStatusCode.PAGE_ERROR,
) : BaseException(statusCode) {

  companion object {
    private const val serialVersionUID = -1202049169478818925L
  }
}